protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -824,26 +850,38 @@ func templateToOpenAPIPath(path string, reg *descriptor.Registry, fields []*desc
 	// syntax for this subsection CAN be handled by a regexp since it has no
 	// memory.
 	for index, part := range parts {
-		// If part is a resource name such as "parent", "name", "user.name", the format info must be retained.
-		prefix := canRegexp.ReplaceAllString(part, "$1")
-		if isResourceName(prefix) {
+		if !pathParamTemplateRegex.MatchString(part) {
+			continue
+		}
+
+		if !(strings.HasPrefix(part, "{") && strings.Contains(part, "=") && strings.Contains(part, "}")) {
 			continue
 		}
-		parts[index] = canRegexp.ReplaceAllString(part, "{$1}")
+
+		submatch := pathParamTemplateRegex.FindStringSubmatch(part)
+		paramName := submatch[1]
+		paramTemplate := submatch[2]
+		rest := submatch[3]
+
+		if strings.Count(paramTemplate, "*") <= 1 {
+			parts[index] = strings.Replace(paramTemplate, "*", fmt.Sprintf("{%s}", paramName), 1) + rest
+		} else {
+			templateParts := strings.Split(paramTemplate, "/")
+			var previousPart string
+			for i, templatePart := range templateParts {
+				if templatePart == "*" {
+					templateParts[i] = fmt.Sprintf("{%s.%s}", paramName, previousPart)
+				} else {
+					previousPart = templatePart
+				}
+			}
+			parts[index] = strings.Join(templateParts, "/") + rest
+		}
 	}
 
 	return strings.Join(parts, "/")
 }
 
-func isResourceName(prefix string) bool {
-	words := strings.Split(prefix, ".")
-	l := len(words)
-	field := words[l-1]
-	words = strings.Split(field, ":")
-	field = words[0]
-	return field == "parent" || field == "name"
```

Comment:
Is it safe to remove this?

Reply:
Even after reading the issues and PRs related to this, I still have no idea why it was implemented this way. What's so special about `parent` and `name` other than being the recommended resource names to use by Google for writing APIs and how does that relate to openapi v2? Of course, if there's some convention to follow when creating the openapi v2 parameters based on these resource names, that would be nice to know and probably expected by users.

Reply:
Yea I don't know either. @ivucica do you know?